CMakeLists.txt

```diff
@@ -40,7 +40,7 @@ For more information, please visit <http://www.openshot.org/>.
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/Modules")
 
 ################ PROJECT VERSION ####################
-set(PROJECT_VERSION_FULL "0.1.9-dev1")
+set(PROJECT_VERSION_FULL "0.2.0-dev1")
 set(PROJECT_SO_VERSION 7)
 set(HEX_VERSION_OVERRIDE "0x109") # For CMake < 3.13
```

Comment:
@ferdnyc Any thoughts on if HEX_VERSION_OVERRIDE is still needed? In a related question, I'm having trouble with this syntax:

```
math(EXPR PROJECT_VERSION_HEX
"(${PROJECT_VERSION_MAJOR} << 16) + \
		(${PROJECT_VERSION_MINOR} << 8) + \
		(${PROJECT_VERSION_PATCH})" OUTPUT_FORMAT HEXADECIMAL )
```

With cmake version 3.5.1, it gives me the following error:
```
CMake Error at CMakeLists.txt:59 (math):
  math EXPR called with incorrect arguments.
```

Error goes away when I remove `OUTPUT_FORMAT HEXADECIMAL`, but then it returns a non-hex output.

Reply:
Yeah, unfortunately `OUTPUT_FORMAT_HEXADECIMAL` only appeared in CMake 3.13. Which is why the ugly hardcoding for the `HEX_VERSION_OVERRIDE` is still necessary, unless we can come up with another cross-platform way of generating a hex number from a dotted decimal version string.

Reply:
`HEX_VERSION_OVERRIDE` for `0.2.0` would be `0x200`. (It's really `0x000200`, two hex digits per level, but we drop the leading zeroes.)

Reply:
Oh, 3.13 > 3.5.1... those darn dots

Reply:
Any clever syntax to generate the correct hex version, on systems with older Cmake?

Reply:
Unfortunately no, if I could think of _any_ solution other than hardcoding it like that, I'd definitely have gone with it.

Reply:
Actually, come to think of it I do have _one_ idea. (Or, I just had an idea.) It feels like overkill, _sort of_, but it's really not _that_ crazy and it's probably the only thing that'd work.

CMake has a [`try_run`](https://cmake.org/cmake/help/v3.2/command/try_run.html) command that's existed since long before CMake 3.2, mostly meant for compiler feature detection and whatnot. It takes a source file name and a bunch of compilation parameters, compiles the source into an executable, and then runs it. The success/failure status of the compile and run steps are both reported, but more importantly, the **OUTPUT** of both steps can also be captured.

So, what we could do is plug `${PROJECT_VERSION_[MAJOR|MINOR|BUILD]}` into a `GENERATED` source file (written at generation time using `configure_file`) that converts them into hex, then prints the result to stdout. If we have CMake `try_run()` that program, and use the `RUN_OUTPUT_VARIABLE` argument to direct the output somewhere, then voila: we get our hex version without having to hardcode it. And since we know any system building `libopenshot-audio` has a working C/C++ compiler...

...Actually, scratch that, we wouldn't even need to generate the file. (Which is good, because I'm not sure that'd work — the file might not be generated before the `try_run()` is executed.) `try_run()` takes a `COMPILE_DEFINITIONS` argument, so we can just write the source to use `PROJECT_VERSION_` variables in place of the actual values, and set them at compile time using `COMPILE_DEFINITIONS`.

Reply:
Done, in #89. `HEX_VERSION_OVERRIDE` is dead, long live `PROJECT_VERSION_HEX`.